<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>bis</class>
 <widget class="QMainWindow" name="bis">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>580</width>
    <height>660</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>580</width>
    <height>660</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>580</width>
    <height>660</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Ecrypter</string>
  </property>
  <property name="windowOpacity">
   <double>1.000000000000000</double>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <property name="locale">
   <locale language="Russian" country="Russia"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLineEdit" name="lineSource">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>30</y>
      <width>200</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>the treasure is buried near the house</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnCrip">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>210</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Cript</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnUnCrip">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>210</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>UnCript</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineCript">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>170</y>
      <width>200</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineUncript">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>170</y>
      <width>221</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="labelSL">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Source line</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelKey">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>80</y>
      <width>47</width>
      <height>14</height>
     </rect>
    </property>
    <property name="text">
     <string>Key</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineKey">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>70</y>
      <width>200</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>cipher</string>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>30</y>
      <width>221</width>
      <height>92</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QCheckBox" name="checkHoriz">
       <property name="text">
        <string>Горизонтальный сдвиг</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
       <property name="autoExclusive">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="checkVij">
       <property name="text">
        <string>Виженер</string>
       </property>
       <property name="autoExclusive">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="checkAutoKey">
       <property name="text">
        <string>AutoKey</string>
       </property>
       <property name="autoExclusive">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="checkFeist">
       <property name="text">
        <string>Feist</string>
       </property>
       <property name="autoExclusive">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="btnClear">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>210</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelCriptedSL">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>150</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Защифрованное сообщение</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelDecriptedSL">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>150</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Восстановленное сообщение</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnDihe_A">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>410</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Dihe open key A</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineDihe_A">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>370</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineDihe_g">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>310</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineDihe_p">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>370</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineDihe_B">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>370</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="btnDihe_B">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>410</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Dihe open key B</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineDihe_key">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>340</y>
      <width>221</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="labelOpenA">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>350</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Open A</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelOpenB">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>350</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Open B</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelOpenG">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>290</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Open G</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelOpenP">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>350</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Open P</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelKey_DiHe">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>320</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Key</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnDihe_key_from_a">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>380</y>
      <width>70</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Key ftom A</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="btnDihe_Key_From_B">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>380</y>
      <width>70</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Key from b</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="btnDihe_Clear">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>380</y>
      <width>60</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>240</y>
      <width>580</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>215</x>
      <y>250</y>
      <width>140</width>
      <height>30</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: italic 18pt &quot;Brush Script MT&quot;;
</string>
    </property>
    <property name="text">
     <string>Dieffie-Hellman</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineDihe_Hide_A">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>310</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineDihe_Hide_B">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>310</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="labelHideA_DiHe">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>290</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Hide A</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelHideB_DiHe">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>290</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Hide B</string>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>440</y>
      <width>580</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineRSA_Open_A">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>520</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="btnRSA_Open_A">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>560</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>RSA open key A</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelOpenA_RSA">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>500</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Open A</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnRSA_Open_B">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>550</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Cripted text to A</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineRSA_Source">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>500</y>
      <width>241</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>9</string>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>235</x>
      <y>460</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 18pt &quot;Ravie&quot;;
</string>
    </property>
    <property name="text">
     <string>RSA</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelSL_RSA">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>480</y>
      <width>141</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Source line</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineRSA_Cripted">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>520</y>
      <width>201</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineRSA_Decripted">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>590</y>
      <width>201</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="labelCripted_RSA">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>500</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Защифрованное сообщение</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelDecripted_RSA">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>570</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Восстановленное сообщение</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnRSA_Decript_A">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>590</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Decripted from B</string>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>450</y>
      <width>20</width>
      <height>181</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>580</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>Options</string>
    </property>
    <widget class="QMenu" name="menuChange_lang">
     <property name="title">
      <string>Change lang</string>
     </property>
     <addaction name="actionRussian"/>
     <addaction name="actionEnglish"/>
    </widget>
    <addaction name="menuChange_lang"/>
    <addaction name="actionDebug_info"/>
   </widget>
   <addaction name="menuOptions"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionRussian">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Russian</string>
   </property>
   <property name="autoRepeat">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionEnglish">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>English</string>
   </property>
   <property name="autoRepeat">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionDebug_info">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Debug info</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
